pkg/k8shandler/deployment.go

```diff
@@ -30,41 +30,39 @@ func (node *deploymentNode) isDifferent(cfg *desiredNodeState) (bool, error) {
 	// Check replicas number
 	actualReplicas := *node.resource.Spec.Replicas
 	if cfg.getReplicas() != actualReplicas {
-		logrus.Debugf("Different number of replicas detected, updating deployment %v", cfg.DeployName)
+		logrus.Debugf("Different number of replicas detected, updating deployment %q", cfg.DeployName)
 		return true, nil
 	}
 
-	// Check image of Elasticsearch container
-	for _, container := range node.resource.Spec.Template.Spec.Containers {
-		if container.Name == "elasticsearch" {
-			if container.Image != cfg.ESNodeSpec.Spec.Image {
-				logrus.Debugf("Resource '%s' has different container image than desired", node.resource.Name)
-				return true, nil
-			}
-		}
-	}
-
 	// Check if labels are correct
+	if len(cfg.Labels) != len(node.resource.Labels) {
+		logrus.Debugf("Different labels detected, updating deployment %q", node.resource.GetName())
+		return true, nil
+	}
 	for label, value := range cfg.Labels {
 		val, ok := node.resource.Labels[label]
 		if !ok || val != value {
-			logrus.Debugf("Labels on deployment '%v' need update..", node.resource.GetName())
+			logrus.Debugf("Different labels detected, updating deployment %q", node.resource.GetName())
 			return true, nil
 		}
 	}
 
 	// Check if the Variables are the desired ones
 	envVars := node.resource.Spec.Template.Spec.Containers[0].Env
 	desiredVars := cfg.EnvVars
+	if len(envVars) != len(desiredVars) {
+		logrus.Debugf("Different environmental variables detected, updating deployment %q", node.resource.GetName())
+		return true, nil
+	}
 	for index, value := range envVars {
 		if value.ValueFrom == nil {
 			if desiredVars[index] != value {
```

Comment:
I'm guessing the crash was here?

Reply:
Yes, as described in https://bugzilla.redhat.com/show_bug.cgi?id=1662105